src/main/java/net/sf/jabref/gui/importer/fetcher/OAI2FetcherGUI.java

```diff
@@ -0,0 +1,162 @@
+package net.sf.jabref.gui.importer.fetcher;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Date;
+
+import javax.swing.JOptionPane;
+
+import net.sf.jabref.gui.help.HelpFiles;
+import net.sf.jabref.importer.ImportInspector;
+import net.sf.jabref.importer.OAI2Handler;
+import net.sf.jabref.importer.OutputPrinter;
+import net.sf.jabref.importer.fetcher.OAI2Fetcher;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.IdGenerator;
+import net.sf.jabref.model.entry.MonthUtil;
+
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+public class OAI2FetcherGUI extends OAI2Fetcher implements EntryFetcherGUI {
+
+    @Override
+    public HelpFiles getHelpPage() {
+        return HelpFiles.FETCHER_OAI2_ARXIV;
+    }
+
+    /**
+     * some archives - like ArXiv.org - might expect of you to wait some time
+     */
+    private boolean shouldWait() {
+        return getWaitTime() > 0;
+    }
+
+    /**
+     * Import an entry from an OAI2 archive. The BibEntry provided has to
+     * have the field OAI2_IDENTIFIER_FIELD set to the search string.
+     *
+     * @param key
+     *            The OAI2 key to fetch from ArXiv.
+     * @return The imported BibEntry or null if none.
+     */
+    public BibEntry importOai2Entry(String key) {
+        /**
+         * Fix for problem reported in mailing-list:
```

Comment:
Either you remove this comment or you describe what a problem it was, since the link isn't working